pkg/cloudclient/aws/private.go

```diff
@@ -643,25 +643,72 @@ func (c *Client) deleteARecord(clusterDomain, DNSName, aliasDNSZoneID, resourceR
 	return err
 }
 
+// recordExists checks if a specific RecordSet already exist in route53
+func (c *Client) recordExists(resourceRecordSet *route53.ResourceRecordSet, publicHostedZoneID string) (bool, error) {
+	if resourceRecordSet == nil {
+		return false, goError.New("resourceRecordSet can't be nil")
+	}
+	if resourceRecordSet.Name == nil {
+		return false, goError.New("resourceRecordSet Name is required")
+	}
+	// Route53 assumes that the domain name that you specify is fully qualified.
+	// Therefore, it doesn't mind if the trailing "." is missing.
+	// However, adding the trailing "." make the future comparaison easier, since route53 listResourceRecordSets contains trailing dots
+	if !strings.HasSuffix(*resourceRecordSet.Name, ".") {
+		*resourceRecordSet.Name += "."
+	}
+	if resourceRecordSet.AliasTarget != nil && !strings.HasSuffix(*resourceRecordSet.AliasTarget.DNSName, ".") {
+		*resourceRecordSet.AliasTarget.DNSName += "."
+	}
+
+	var recordExists bool
+	input := &route53.ListResourceRecordSetsInput{
+		HostedZoneId: aws.String(publicHostedZoneID),
+	}
+
+	// Check if any property of the record was changed
```

Comment:
```suggestion
	// Check if any property of the record was changed
	// aws-go-sdk may potentially give all results in multiple pages, so this will go
	//  through every page given in response to the API call
```

Nit: Add a note about pages to match with the following comment in the lambda func to help with reading comprehension.

Reply:
thanks for that